### PR TITLE
Add/announcement modal

### LINKED ISSUE #56955: Marketplace: Create a modal introducing the new Plugins landing page
With the introduction of the Marketplace we are making a few changes in the UI / UX of plugins list, details, and installation flow. Moreover, we will force Atomic users to to the Calypso plugins page when they click "Add New" plugin from the sidebar. We will need to make them aware of these changes and the "why" behind them.

Let's create some content for the modal component we already have in Calypso. This way we can pop it up, once the User has reached the Plugins page and educate them.

- [x] Design
- [x] Development

### PR DESCRIPTION
**Pull request by cpapazoglou**
#### Changes proposed in this Pull Request

* Created a new Announcement modal block that can be reused in future cases
* Added the modal in the plugin browser landing page using a feature flag `marketplace`

#### Testing instructions

* Go to `/plugins`
* You should see the modal
* Click X or 'Let's explore' 
* Refresh the page, you shouldn't see the modal
* If you delete it from your preferences you may see it again 
SS 2021-11-04 at 14 44 09

SS 2021-11-04 at 14 44 42

Fixes #56955

### CODE REVIEW

File: client/blocks/announcement-modal/index.jsx [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+						image={
+							<picture className="announcement-modal__picture" key={ page.featureImage }>
+								<source srcSet={ page.featureImage } media="(min-width: 600px)" />
+								<img alt="" src={ page.featureImage } />
+							</picture>
+						}
+						headline={ page.headline }
+						heading={ page.heading }
+						content={ page.content }
+						cta={ singlePage ? finishButtonText : null }
```
Comment (WBerredo) [inline client/blocks/announcement-modal/index.jsx:70]: Nitpicking: Why exactly the CTA is shown only with single pages?

**Review by WBerredo — APPROVED** [review]
LGTM

File: client/blocks/announcement-modal/style.scss [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+	@media ( min-width: $modal-breakpoint ) {
+		margin-top: 0;
+		width: 90%;
+	}
+}
+
+.announcement-modal__heading {
+	@include onboarding-font-recoleta;
+	/* Gray / Gray 90 */
+	color: #1d2327;
```
Comment (WBerredo) [inline client/blocks/announcement-modal/style.scss]: Nitpicking: Any reason for not using the ` --studio-gray-90` variable?

File: client/blocks/announcement-modal/style.scss [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+		}
+	}
+}
+
+.announcement-modal__description p {
+	font-size: 15px;
+	line-height: 22px;
+
+	/* Gray / Gray 60 */
+	color: #50575e;
```
Comment (WBerredo) [inline client/blocks/announcement-modal/style.scss]: Same as above.

Comment (cpapazoglou) [reply client/blocks/announcement-modal/style.scss]: 46bf8a3

File: client/blocks/announcement-modal/style.scss (hunk 2 above)
Comment (cpapazoglou) [reply client/blocks/announcement-modal/style.scss]: 46bf8a3

File: client/blocks/announcement-modal/index.jsx (hunk 1 above)
Comment (cpapazoglou) [reply client/blocks/announcement-modal/index.jsx:70]: In a multi page modal, a CTA linking to another page may lead to unexpected behaviors. So, I wanted to keep it contained at that moment. We can iterate further when there is a need of having a CTA in multiple pages.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/6944850

Thank you @cpapazoglou for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/6944850).

**Comment by noahtallen** [comment]
:wave: I wanted to link this existing work just in case it wasn't on your radar! https://github.com/Automattic/wp-calypso/pull/48722

I think they are accomplishing very similar things, so it probably makes sense to make sure they are consistent. maybe What's New could be updated to use this new component? https://github.com/Automattic/wp-calypso/tree/trunk/packages/whats-new

**Comment by cpapazoglou** [comment]
> 👋 I wanted to link this existing work just in case it wasn't on your radar! #48722
> 
> I think they are accomplishing very similar things, so it probably makes sense to make sure they are consistent. maybe What's New could be updated to use this new component? https://github.com/Automattic/wp-calypso/tree/trunk/packages/whats-new

Yeah, I have bumped in to `whats-new` while searching for modal implementations. I couldn't reuse that cause it has the content hardcoded, but it seems that it would could be worth trying to make `whats-new` use the new announcement modal by first enhancing it. I guess it should be also moved to `packages/components` so that it gets reused from ETK? It seems that we also have https://github.com/Automattic/wp-calypso/pull/50224

**Comment by noahtallen** [comment]
Yeah, I believe the whats-new package is imported by ETK and then ETK inserts it into the editor. I think there were some plans to make it dynamic, but not sure if that happened. cc @donlair

**Comment by noahtallen** [comment]
either way, it'd be lovely to have a consistent announcements system!

**Comment by donlair** [comment]
@noahtallen @cpapazoglou yep, we've got an MC tool that's documented at the `whats-new-feature-announcement-tool` page in FG. 

The endpoint was deployed in D56831-code and the What's New guide was converted to that pull data from the endpoint in [#50224](https://github.com/Automattic/wp-calypso/pull/50224).

**Pull request merged**